src/go/k8s/pkg/resources/configmap.go

```diff
@@ -681,14 +711,39 @@ func (r *ConfigMapResource) GetLastAppliedConfigurationFromCluster(
 	}
 	if ann, ok := existing.Annotations[LastAppliedConfigurationAnnotationKey]; ok {
 		var cnf map[string]interface{}
-		if err := json.Unmarshal([]byte(ann), &cnf); err != nil {
+		decoder := json.NewDecoder(bytes.NewReader([]byte(ann)))
+		decoder.UseNumber()
+		if err := decoder.Decode(&cnf); err != nil {
 			return nil, false, fmt.Errorf("could not unmarshal last applied configuration from configmap annotation %q: %w", LastAppliedConfigurationAnnotationKey, err)
 		}
+		cleanupConfig(cnf)
 		return cnf, true, nil
 	}
 	return nil, false, nil
 }
 
+// cleanupConfig translates json Number objects into int64, otherwise yaml.v3
```

Comment:
I'm not sure of the context where this is used, but in general it isn't correct to assume all numbers are integers: floating point configuration properties do exist.  The config schema distinguishes between `number` and `integer` types.

Reply:
Yes, I've noticed the presence of some floating point properties and fixed this later.